ui/app/AppLayouts/Onboarding2/pages/LoginScreen.qml

```diff
@@ -80,44 +115,18 @@ OnboardingPage {
         }
     }
 
-    Component.onCompleted: {
-        loginUserSelector.setSelection(d.settings.lastKeyUid)
-        if (!d.currentProfileIsKeycard)
-            passwordBox.forceActiveFocus()
-    }
-
-    function setObtainingPasswordError(error: string, wrongFingerprint: bool) {
-        d.biometricsSuccessful = false
-        d.biometricsFailed = wrongFingerprint
-
-        if (d.currentProfileIsKeycard) {
-            keycardBox.clear()
-        } else {
-            passwordBox.validationError = error
-            passwordBox.clear()
-            passwordBox.forceActiveFocus()
-        }
-    }
-
-    function setObtainingPasswordSuccess(password: string) {
-        if (!root.isBiometricsLogin)
+    onKeycardStateChanged: {
+        if (!biometricsAvailable || !isBiometricsLogin || !d.currentProfileIsKeycard
+                || root.keycardState !== Onboarding.KeycardState.NotEmpty)
```

Comment:
I think we need to also compare the selected `KeyUID` with the `KeyUID` on the keycard here?
I mean in case of a wrong keycard inserted.